apidef/oas/operation.go

```diff
@@ -620,51 +622,116 @@ type pathPart struct {
 	name    string
 	value   string
 	isRegex bool
+	isParam bool
 }
 
 func (p pathPart) String() string {
-	if p.isRegex {
+	if p.isRegex || p.isParam {
 		return "{" + p.name + "}"
 	}
 
 	return p.value
 }
 
-// isRegex checks if value has expected regular expression patterns.
-func isRegex(value string) bool {
-	for _, pattern := range regexPatterns {
-		if strings.Contains(value, pattern) {
-			return true
-		}
-	}
-	return false
-}
-
 // splitPath splits URL into folder parts, detecting regex patterns.
 func splitPath(inPath string) ([]pathPart, bool) {
 	// Each URL fragment can contain a regex, but the whole
 	// URL isn't just a regex (`/a/.*/foot` => `/a/{param1}/foot`)
-	parts := strings.Split(strings.Trim(inPath, "/"), "/")
+	inPath = strings.Trim(inPath, "/")
+
+	if inPath == "" {
+		return []pathPart{}, false
+	}
+
+	if err := validatePath(inPath); err != nil {
+		return []pathPart{}, false
+	}
+
+	parts := strings.Split(inPath, "/")
+
 	result := make([]pathPart, len(parts))
 	found := 0
+	nCustomRegex := 0
 
 	for k, value := range parts {
-		name := value
-		isRegex := isRegex(value)
-		if isRegex {
+		// Handle non-bracketed path segments
+		// for example: /a/b/c, /a/[0-9]
+		if !isMuxTemplate(value) {
+			name := value
+
+			result[k] = pathPart{
+				name:  name,
+				value: value,
+			}
+
+			if isRegex(value) {
+				nCustomRegex++
+				found++
+
+				result[k].name = fmt.Sprintf("customRegex%d", nCustomRegex)
+				result[k].isRegex = true
+			}
+
+			continue
+		}
+
+		// if !isValidPathSegment(value) {
+		// 	continue
+		// }
+
+		// Handle bracketed path segments
+		segment := strings.Trim(value, "{}")
+
+		// Parameter with pattern case:
+		// for example: /a/{id:[0-9]}
+		if name, pattern, ok := strings.Cut(segment, ":"); ok && strings.TrimSpace(name) != "" {
+			// this is required because GetPathRegexp doesn't check if the regex is valid
+			// it doesn't return an error even if the regex is invalid (eg. [a-zA-Z+)
+			if _, err := regexp.Compile(pattern); err != nil {
```

Comment:
change the import to this to the /regexp package, caching the compilation

Reply:
Done